src/sources/nats.rs

```diff
@@ -132,29 +132,30 @@ async fn nats_source(
     let stream = get_subscription_stream(subscription).take_until(shutdown);
     pin_mut!(stream);
     while let Some(msg) = stream.next().await {
+        emit!(&BytesReceived {
+            byte_size: msg.data.len(),
+            protocol: "tcp",
+        });
         let mut stream = FramedRead::new(msg.data.as_ref(), decoder.clone());
         while let Some(next) = stream.next().await {
             match next {
                 Ok((events, byte_size)) => {
-                    emit!(&NatsEventsReceived {
-                        byte_size,
-                        count: events.len()
-                    });
+                    let count = events.len();
+                    emit!(&NatsEventsReceived { byte_size, count });
```

Comment:
I don't think the `byte_size` here is right. The `byte_size` returned by the decoder is actually the number of bytes from the frame, so basically the same as received bytes above that is published as `component_received_bytes_total`:

https://github.com/vectordotdev/vector/blob/86bfec8d70cb30158964a0208d37b30723be3e8f/src/codecs/decoding.rs#L191-L216

Instead, `component_received_event_bytes_total` should reflect the total deserialized event bytes in-memory. That is, you should sum up `ByteSizeOf::size_of` of the events here.